tests/patmat/i4225b.scala

```diff
@@ -0,0 +1,12 @@
+object Bar {
+  def unapply(x: String): Some[Int] =
+    Some(0)
+}
+
+object Test {
+  def test(x: String) =
+    x match {
+      case Bar(a) => a
+      case _ => x // this case is reachable, i.e. test(null)
```

Comment:
Suggestion; maybe if the _only_ way to reach a case is through null, we should suggest the user writes `case null` instead of `case _` ?

Reply:
This is a good idea, it's implemented in the latest commit.